.azure-pipelines/generate-beta-typings.yml

```diff
@@ -0,0 +1,166 @@
+  inputs:
+    targetType: inline
+    script: |
+      $repoModelsDir = "$env:Build_SourcesDirectory\msgraph-typescript-typings\"
+      Write-Host "Path to repo model directory: $repoModelsDir"
+      Write-Host "##vso[task.setvariable variable=repoModelsDir]$repoModelsDir"
+
+      $outputPath = Join-Path $env:Build_SourcesDirectory "output"
+      Write-Host "Path to typewriter.exe output $outputPath"
+      Write-Host "##vso[task.setvariable variable=outputPath]$outputPath"
+
+      $cleanMetadata = "https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/clean_beta_metadata/cleanMetadataWithDescriptionsbeta.xml"
+      Write-Host "Path to clean metadata $cleanMetadata"
+      Write-Host "##vso[task.setvariable variable=cleanMetadata]$cleanMetadata"
+
+      $branchName = "pipelinebuild/beta/$env:Build_BuildId" # Match the spec in the GH Action
+      Write-Host "Branch path spec for the pull request will be $branchName"
+      Write-Host "##vso[task.setvariable variable=branchName]$branchName"
+
+- task: PowerShell@2
+  displayName: 'Git: checkout beta'
+  inputs:
+    targetType: inline
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-typescript-typings'
+    script: |
+      git checkout beta | Write-Host
+
+- task: PowerShell@2
+  displayName: 'Git: branch from beta named with the build id: $(Build.BuildId)'
+  inputs:
+    targetType: inline
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-typescript-typings'
+    script: |
+      Write-Host "The new branch name will be: $env:branchName"
+      git checkout -B $env:branchName | Write-Host
+
+- task: PowerShell@2
+  displayName: 'Git: set user config'
+  inputs:
+    targetType: inline
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-typescript-typings'
+    script: |
+      git config user.email "GraphTooling@service.microsoft.com"
+      git config user.name "Microsoft Graph DevX Tooling"
+
+- task: PowerShell@2
+  displayName: 'Remove generated typings from the repo'
+  inputs:
+    targetType: inline
+    script: |
+      $filePath = Join-Path $env:repoModelsDir "microsoft-graph.d.ts"
+      Remove-Item $filePath | Write-Host
+      Write-Host "Removed the existing typings files in the repo." -ForegroundColor Green
+
+- task: PowerShell@2
+  displayName: 'Typewriter: generate beta typings files'
+  inputs:
+    targetType: filePath
+    filePath: '$(Build.SourcesDirectory)/msgraph-metadata/scripts/runTypewriter.ps1'
+    arguments: '-verbosity Info -language TypeScript -metadata $(cleanMetadata) -output $(outputPath) -generationMode Files -p typescript.namespacePostfix:beta'
+    workingDirectory: '$(Build.SourcesDirectory)' # Set the root for a multi-repo pipeline. /s
+  enabled: true
+
+- task: PowerShell@2
+  displayName: 'Copy generated .d.ts into the repo'
+  inputs:
+    targetType: inline
+    script: |
+      $modelsDirectory = Join-Path $env:outputPath "\com\microsoft\graph\src\microsoft-graph.d.ts"
```

Comment:
Rename it to `$generatedModelsFile` given that this is not a directory.

Reply:
Done